pkg/controllers/vdb/uninstall_reconciler.go

```diff
@@ -121,31 +123,16 @@ func (s *UninstallReconciler) uninstallPodsInSubcluster(ctx context.Context, sc
 	startPodIndex, endPodIndex int32) (ctrl.Result, error) {
 	podsToUninstall, requeueNeeded := s.findPodsSuitableForScaleDown(sc, startPodIndex, endPodIndex)
 	if len(podsToUninstall) > 0 {
-		basePod, err := findATBasePod(s.Vdb, s.PFacts)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-		ipsToUninstall := []string{}
-		for _, p := range podsToUninstall {
-			ipsToUninstall = append(ipsToUninstall, p.podIP)
-		}
-		atConfTempFile, err := s.ATWriter.RemoveHosts(ctx, basePod, ipsToUninstall)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-		defer os.Remove(atConfTempFile)
+		var result ctrl.Result
+		var err error
 
-		if err := distributeAdmintoolsConf(ctx, s.Vdb, s.VRec, s.PFacts, s.PRunner, atConfTempFile); err != nil {
-			return ctrl.Result{}, err
+		if vmeta.UseVClusterOps(s.Vdb.Annotations) {
+			result, err = s.uninstallPodsInSubclusterForVClusterOps(ctx, podsToUninstall)
+		} else {
+			result, err = s.uninstallPodsInSubclusterForAdmintools(ctx, podsToUninstall)
```

Comment:
I think we should move this behind the vadmin/dispatcher abstraction. I would like all admintools stuff only in that package. I know we don't do that for install, but it's something I would like to change too. vadmin can't be given a PodFact, but I think we only need the IP and pod name from them, so hopefully it isn't too bad to create options for those.

Reply:
We need more than just ip and pod name for admintools. There are some functions like `distributeAdmintoolsConf()` or `findATBasePod()` that depend on PodFacts. Given that vadmin can't be given a PodFact, in order to move the whole thing under vadmin/dispatcher abstraction, we must refactor and move those functions to vadmin/at.go and has to be done simultaneously on `install` as it shares some of those functions with `uninstall`.

It can be done on this PR but I personally suggest to do it separately.

Reply:
Yes I will open a ticket.